src/Cardano/Wallet/Primitive/Types.hs

```diff
@@ -294,28 +303,32 @@ instance Buildable (TxIn, TxOut) where
     build (txin, txout) = build txin <> " ==> " <> build txout
 
 data TxMeta = TxMeta
-    { txId :: !(Hash "Tx")
-    , depth :: !(Quantity "block" Natural)
+    { txMetaId :: !(Hash "Tx")
     , status :: !TxStatus
     , direction :: !Direction
-    , timestamp :: !Timestamp
     , slotId :: !SlotId
-    } deriving (Show, Eq, Generic)
+    } deriving (Show, Eq, Ord, Generic)
+
+instance NFData TxMeta
```

Comment:
@rvl Not sure why the instance was defined by-hand for TxMeta :thinking: ? I've changed that to rely on the default instance (which uses the generic representation). If there was a particular reason not to, I'd suggest to attach a comment explaining why ?

Reply:
It was just a workaround